pkg/querier/batch/merge_test.go

```diff
@@ -61,3 +63,47 @@ func TestMergeHarder(t *testing.T) {
 		})
 	}
 }
+
+// TestBugInMergeIterator tests a bug while calling Seek() on mergeIterator.
+func TestBugInMergeIteratorSeek(t *testing.T) {
```

Comment:
nit on naming: aren't all tests testing for a bug? 
```suggestion
// TestMergeIterator tests a bug while calling Seek() on mergeIterator.
func TestMergeIteratorSeek(t *testing.T) {
```